File: backend/src/main/java/com/inventory/backend/repository/RefillHistoryRepository.java
package com.inventory.backend.repository;

import com.inventory.backend.entity.RefillHistory;
import com.inventory.backend.entity.RefillStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface RefillHistoryRepository extends JpaRepository<RefillHistory, Long> {

    List<RefillHistory> findByCartridgeIdOrderByIdDesc(Long cartridgeId);

    Optional<RefillHistory> findFirstByCartridgeIdAndStatusOrderByIdDesc(Long cartridgeId, RefillStatus status);
}

File: backend/src/main/kotlin/com/inventory/backend/repository/RefillHistoryRepository.kt
package com.inventory.backend.repository

import com.inventory.backend.entity.RefillHistory
import com.inventory.backend.entity.RefillStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface RefillHistoryRepository : JpaRepository<RefillHistory, Long> {
    fun findByCartridgeIdOrderByIdDesc(cartridgeId: Long): List<RefillHistory>
    fun findFirstByCartridgeIdAndStatusOrderByIdDesc(cartridgeId: Long, status: RefillStatus): Optional<RefillHistory>
}
